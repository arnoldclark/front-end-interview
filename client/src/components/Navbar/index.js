import React from "react";
import { Link } from "react-router-dom";

export const Navbar = () => (
  <header className="ch-navbar">
    <div className="ch-container">
      <div className="ch-row">
        <div className="xs:ch-col--12 sm:ch-col--3">
          <Link to="/" className="ch-navbar__title">
            Arnold Clark Garage
          </Link>
        </div>
        <div className="xs:ch-col--12 sm:ch-col--9">
          <ul className="ch-navbar__nav" data-id="ch-navbar-menu">
            <li>
              <Link to="/branches">Branch List</Link>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </header>
);
