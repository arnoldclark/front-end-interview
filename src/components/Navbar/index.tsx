export default function Navbar({ title = 'Arnold Clark Garage' }) {
  return (
    <header className="ch-navbar">
      <div className="ch-container">
        <div className="ch-row">
          <div className="xs:ch-col--12 sm:ch-col--3">
            <a href="/" className="ch-navbar__title">
              {title}
            </a>
          </div>
        </div>
      </div>
    </header>
  )
}
