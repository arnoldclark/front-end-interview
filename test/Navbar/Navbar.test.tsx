import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import Navbar from '@/components/Navbar'


const PROPS = {
  title: 'hello there'
}

describe('Navbar', () => {

  test('matches snapshot', () => {
    const { container } = render(<Navbar {...PROPS} />)
    expect(container).toMatchSnapshot()
  })

  test('displays title', async () => {
    render(<Navbar {...PROPS} />)

    expect(screen.getByRole('link')).toHaveTextContent(PROPS.title)
  })

  test('displays default title if no title provided', async () => {
    render(<Navbar />)

    expect(screen.getByRole('link')).toHaveTextContent("Arnold Clark Garage")
  })

})