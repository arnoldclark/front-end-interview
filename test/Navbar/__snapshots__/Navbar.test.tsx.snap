// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches snapshot 1`] = `
<div>
  <header
    class="ch-navbar"
  >
    <div
      class="ch-container"
    >
      <div
        class="ch-row"
      >
        <div
          class="xs:ch-col--12 sm:ch-col--3"
        >
          <a
            class="ch-navbar__title"
            href="/"
          >
            hello there
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;
